### ISSUE TITLE
Add image preview in boxes, and copy-pasting images into individual thoughts

### ISSUE THREAD

**Issue by creikey** [juzzlin/Heimer#123]
E.x:
![image](https://user-images.githubusercontent.com/28424206/101086050-298ce280-3565-11eb-9e51-0ba802bf7547.png)

**Comment by juzzlin** [comment]
Hi,

Currently you can attach images by right-clicking on the node and choosing "Attach image". It's a bit different as it currently show the attached image as a background image that is resized based on the node size.

Could you explain a bit what you mean by "copy-pasting images into individual thoughts" :)

**Comment by creikey** [comment]
@juzzlin I would like to associate each message with an image by pasting images from google or screenshot capturing programs. Setting an image as the background of a node does not provide enough space to see the image, while also making the text harder to read.

**Comment by juzzlin** [comment]
Ok, right. Maybe that could indeed be improved.

**Comment by l3td33r** [comment]
How about adding a new node type instead?
Image node, without possibility to add text to it, just displaying image. I believe this would be more in the spirit of mind maps and more intuitive.

Things that would need to be added:
- Node types (?), possibly related to #81 if submaps are going to be another node type.
- Image nodes, connecting them with normal nodes.
- Menu button to load image + shortcut. After loading image, automatically create an image node, with size exact to that of the image resolution.
- Node handle/button to resize the node (while keeping the ratio of width/height).
- Node menu button (right-click menu) to reset image to its original size.

Buttons to change background/text color are not needed.

**Comment by l3td33r** [comment]
> I'm starting to like the idea of special image etc nodes. It could also work so that double-clicking on the image node would open a bigger image viewer.

Yeah, another useful special node type would be "Reference node" that displays the contents of another node (text+image) and updates itself whenever the referenced node is updated. To differentiate it from other nodes, its color can be different, for example 50% transparency. 
Another special property: click on a reference move to move to its source node.

It would be useful for large mind maps when I want to relate nodes in different parts of the map to the same "category" or similar. I can't link them all to one place because the distances are too great and it starts lagging, and also because it isn't easily visible as connected to these nodes. I just create multiple instances of the same node right now which is a waste, especially if I ever need to modify it.

But that's for another issue I suppose. I'm going to merge these ideas into a new one.

I'm already at it so I can mention another idea - a special node that "Groups" nodes. You can put multiple nodes in it and it resizes automatically. If you move it, all its contents move as well.

Concept art below:
![concept art heimer reference and group nodes](https://user-images.githubusercontent.com/75885440/139508272-5bf7b1b4-26a5-488e-896b-cb4f772eb30d.png)
